Trappist/src/Promact.Trappist.Web/wwwroot/app/tests/test-settings/test-settings.component.ts

```diff
@@ -1,65 +1,127 @@
+    /**
+     * Gets the Id of the Test from the route and fills the Settings saved for the selected Test in their respective fields
+     */
     ngOnInit() {
         this.testId = this.route.snapshot.params['id'];
         this.getTestSettings(this.testId);
     }
 
+    /**
+     * Open snackbar
+     * @param message contains the message to be displayed when the snackbar gets opened
+     */
+    openSnackBar(message: string) {
+        let snackBarRef = this.snackbarRef.open(message, 'Dismiss', {
+            duration: 4000,
+        });
+    }
+
     //Gets the Settings saved for a particular Test
-    getTestSettings(id:number) {
-        this.testSettingService.getSettings(id).subscribe((response) => {
+    /**
+     * Gets the Settings saved for a particular Test
+     * @param id contains the value of the Id from the route
+     */
+    getTestSettings(id: number) {
+        this.testService.getTestSettings(id).subscribe((response) => {
             this.testsettings = (response);
         });
     }
 
-    //Selects the Test Name from the text box containing it on focus 
+    /**
+     * Selects the Test Name from the text box containing it on focus
+     * @param $event is used to select the contents of the target text box
+     */
     selectAllContent($event: any) {
         $event.target.select();
-    }   
+    }
 
-    //Updates the Edited Test Name in the Database on removing focus
-    onBlurMethod(id: number, testObject: Test) {
-        this.testSettingService.updateSettings(id, testObject).subscribe((response) => {       
+    /**
+     * Updates the Edited Test Name in the Database om removing focus
+     * @param id contains the value of the Id from the route
+     * @param testObject is an object of class Test
+     * @param value contains the Test Name of the Test which is to be edited 
+     */
+    updateTestName(id: number, testObject: Test, value: string) {
+        this.testService.updateTestSettings(id, testObject).subscribe((response) => {
+            this.openSnackBar(this.testNameUpdationMessage);
         });
     }
 
-    //Launches the Test Dialog Box and also updates the Settings edited for the selected Test
+    /**
+     * Checks the End Date and Time is valid or not
+     * @param endDate contains ths the value of the field End Date and Time
+     */
+    isEndDateValid(endDate: Date) {
+        this.testsettings.startDate > endDate ? this.validEndDate = true : this.validEndDate = false;
+    }
+
+    /**
+     * Checks whether the Start Date selected is valid or not
+     */
+    isStartDateValid() {
+        new Date(this.testsettings.startDate) < this.currentDate || this.testsettings.startDate > this.testsettings.endDate ? this.validStartDate = true : this.validStartDate = false;
+    }
+
+    /**
+     * Checks whether the Warning Time set is valid
+     */
+    validWarningTimeChecking() {
```

Comment:
I have given you name change for above two methods and not given for this method. So these method is as it is. having improper name after giving suggestions.